### PR TITLE
Add quick section on tips to debug UI tests

### PR DESCRIPTION
**Pull request by diosmosis**
### Description:

These aren't really official steps since they are a bit hacky. But from recent conversations about failing UI tests it seems this knowledge is not common or easily derived. They speed up diagnosis and resolution considerably for me.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: docs/debugging-core.md [hunk 1]
```diff
@@ -96,6 +96,15 @@ always_archive_data_day = 0;
 always_archive_data_range = 0;
 ```
 
+### Debugging UI tests
+
+To debug a failing UI test that also fails locally, these steps might be helpful:
+
+1. In global.ini.php (temporarily) set `[Development] enabled = 1` and `[Development] disable_merged_assets = 1` (this allows the UI tests to use unmerged JS so you get better stack traces).
```
Comment (michalkleiner) [inline docs/debugging-core.md:103]: Is `config.ini.php` not respected in the tests?

Comment (diosmosis) [reply docs/debugging-core.md:103]: no, then the results would differ based on your local config

Comment (michalkleiner) [reply docs/debugging-core.md:103]: Interesting. Thanks for the confirmation. Perhaps we could honour tests.ini.php if present to provide test overrides, but that's not relevant here at the moment.

Comment (diosmosis) [reply docs/debugging-core.md:103]: there is `config/environment/*.php`, fyi

File: docs/debugging-core.md [hunk 2]
```diff
@@ -96,6 +96,15 @@ always_archive_data_day = 0;
...
+2. Go into the UI test and put a return after the first test that fails (this speeds up the iteration cycle since you don't have to wait for the entire test to run to get a result).
+3. When the test fails, there will be a 'URL to reproduce' link, open that in a browser.
+4. Debug as if you were developing locally by changing files and reloading. You may have to run `vue:build --watch {Plugin}` for the vue changes.
```
Comment (michalkleiner) [inline docs/debugging-core.md:106]:
```suggestion
4. Debug as if you were developing locally by changing files and reloading. You may have to run `vue:build --watch {Plugin}` for the vue changes.
5. When finished, don't forget to revert the temporarily changed config values from step  1.
```

**Review by michalkleiner — APPROVED** [review]
I'm happy with this once we add the 5th step reminding to reset the config changes.

File: docs/debugging-core.md [hunk 3]
```diff
@@ -96,6 +96,16 @@ always_archive_data_day = 0;
...
+5. When finished, don't forget to revert the temporarily changed config values from step  1.
```
Comment (michalkleiner) [inline docs/debugging-core.md]:
```suggestion
5. When finished, don't forget to revert the temporarily changed config values from step 1.
```

**Pull request merged**